### PR TITLE
fixes missing-prototypes reports

### PR DESCRIPTION
**Pull request by etienne-lms**
Few changes in framework and some modules to address build issues reported when building ith OP-TEE, its build environment and toolchain. I have not covered all modules, only those currently embedded in the optee-* products.

### CODE REVIEW

**Comment by etienne-lms** [comment]
Any update on these changes?

**Comment by etienne-lms** [comment]
I would think that it is better for scp-firmware to declare exported prototypes and more generally follow the toolchains default build warnings recommendation (which evolves with toolchains), unless a good reason prevents that a warning to be addressed.

File: module/scmi_system_power_req/test/mod_scmi_system_power_req_extra.h [hunk 1]
```diff
@@ -9,19 +9,6 @@
  */
 #include <mod_scmi_system_power_req.h>
 
-/*! Send an SCMI message */
```
Comment (nicola-mazzucato-arm) [inline module/scmi_system_power_req/test/mod_scmi_system_power_req_extra.h:12]: these need to stay as they're part of the tests

**Review by nicola-mazzucato-arm — CHANGES_REQUESTED** [review]
Can you please share which compiler you are using and is it used only locally or is there automation in place?
Thanks

**Review by etienne-lms — COMMENTED** [review]
Thanks for the feedback. I'll update this P-R.

File: module/scmi_system_power_req/test/mod_scmi_system_power_req_extra.h (hunk 1 above)
Comment (etienne-lms) [reply module/scmi_system_power_req/test/mod_scmi_system_power_req_extra.h:12]: Discard my comment. This is a test related header file, so fine. I'll discard this change.

Comment (etienne-lms) [reply module/scmi_system_power_req/test/mod_scmi_system_power_req_extra.h:12]: Hmm.. I reconsider my position. 

These 2 prototypes are declared from 2 different header files (here and _mod_scmi_sensor_req_extra.h_). I don't think it is nice. There should be 1 declaration for all sources.
Couldn't _mod_scmi_system_power_req_extra.h_ include _mod_scmi.h_ to get the test setup happy?

**Review by etienne-lms — COMMENTED** [review]
> Can you please share which compiler you are using and is it used only locally or is there automation in place?

As for automation, OP-TEE CI currently only test builds with SCP-firmware/SCMI stack. There is no effective test execution yet. SCP-fmw/OP-TEE is built for 3 platform flavors. A [Qemu/armv8-A](https://github.com/OP-TEE/optee_os/blob/master/.github/workflows/ci.yml#L130) platform is built in Aarch64 and is quite used in OP-TEE for development/tests/demo sharing. OP-TEE CI runs regression tests with a Qemu/armv8 image but these tests currently don't use/cover SCMI resources. CI also builds SCP-fmw for a [FVP armv8-A](https://github.com/OP-TEE/optee_os/blob/master/.github/workflows/ci.yml#L139) platform and an [STM32MP15 armv7-A](https://github.com/OP-TEE/optee_os/blob/master/.github/workflows/ci.yml#L130) platform both in 32bit. 

If i'm right, CI build environment runs in a virtualized x86/ubuntu container. On x86 build hosts OP-TEE currently uses Arm's GCC 10.2 release, for 32 and 64bit ([_toolchain.mk:46_](https://github.com/OP-TEE/build/blob/3.20.0/toolchain.mk#L46-L55)).
OP-TEE distribs can also build on Aarch64 host machines. In that case we use Arm GCC 10.2 cross toolchain for arm 32bit ([toolchain.mk:86](https://github.com/OP-TEE/build/blob/master/toolchain.mk#L86-L89)) but a Buildroot based toolchain for native Aarch64 ([toolchain.mk:91](https://github.com/OP-TEE/build/blob/master/toolchain.mk#L91-L97)).  There are (crazy?) people building with clang/llvm and one could select it's own toolchain (assuming it's reliable) to build OP-TEE. OP-TEE may one day dump to GCC 11 or 12, or support a range of GCC's and Clang's.

(@jforissier, correct me if i'm wrong)

**Comment by etienne-lms** [comment]
I think all review comments have been addressed.

File: module/scmi_system_power_req/test/mod_scmi_system_power_req_extra.h (hunk 1 above)
Comment (girishpathak) [reply module/scmi_system_power_req/test/mod_scmi_system_power_req_extra.h:12]: @etienne-lms : can we move to a separate header mod_scmi_ut.h ? and keep this file under module/scmi/test/includes ? 
scmi_sensor and system power test modules CMakeLists.txt then can include a path for this header. This way it won't affect mod_scmi.h

**Review by girishpathak — APPROVED** [review]

**Comment by girishpathak** [comment]
Jenkins: ok to test

**Review by nicola-mazzucato-arm — APPROVED** [review]

**Comment by nicola-mazzucato-arm** [comment]
Agreed with @etienne-lms to close this PR. The same request will be then opened in gitlab soon.
We will provide more information about what failed for unit testing.
Thanks

**Pull request closed without merging**